kratos/utilities/parallel_utilities.h

```diff
@@ -0,0 +1,364 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//                   Denis Demidov
+//
+
+#if !defined(KRATOS_PARALLEL_UTILITIES_H_INCLUDED )
+#define  KRATOS_PARALLEL_UTILITIES_H_INCLUDED
+
+
+// System includes
+#include<iostream>
+#include<array>
+#include<vector>
+#include<tuple>
+#include<cmath>
+#include<limits>
+#include<omp.h>
+
+#include <future>
+#include <thread>
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+
+//***********************************************************************************
+//***********************************************************************************
+//***********************************************************************************
+template<class TContainerType, class TIteratorType=typename TContainerType::iterator>
+class BlockPartition
+{
+public:
+
+    static constexpr int max_allowed_threads = 128;
```

Comment:
```suggestion
template<class TContainerType, class TIteratorType=typename TContainerType::iterator,
    int max_allowed_threads=128>
class BlockPartition
{
public:
```
So that the value of `max_allowed_threads` could be changed by the user if necessary.

Reply:
you make a good point, but with a but: i think that the 128 should be taken from a global variable, so that one could change it at once for the whole kratos.

I mean, imagine that in a future we have cpus with 1000 cores. Then we should be able to change this without crawling the entire code. (of course an option is to do it dynamically)